### PR TITLE
Fix #60 #18 Controller mode toggle and controller mappings

### LINKED ISSUE #60: Change controller aiming to the right stick
**Estimated:** 30 **Min:** 30 **Max:** 60

### PR DESCRIPTION
**Pull request by clayton-snyder**
* Added a static variable in ChangeScene.cs (since that's currently the only script in the Menu scene for now; this could go in a dedicated file if we have other options to set from the menu later on) called SelectedControllerMode. Other scripts can access this to tell whether the player selected ControllerMode from the menu.

* Added a button which toggles SelectedControllerMode in the Menu scene.

* Start() in PlayerController.cs changes the axis strings to their "*_P2" versions if ControllerMode is true. ControllerMode is initialized in Start() to match ChangeScene.SelectedControllerMode; it's used enough that it's probably worth keeping the local variable here.

* Start() in RopeSystem.cs does the same for the FireAxis and ClimbDescendAxis string.

* Changed Jump_P2 axis to use the A button.

* Changed Horizontal_P2 and Vertical_P2 axes to use the right analog stick (4th and 5th axes) instead of the left stick.

* Added menu buttons to select mouse or controller mode.

Issue #60 - **Estimated: 30** **Min: 30** **Max: 60** **Actual: 5**
Issue #18 - **Estimated: 60** **Min: 30** **Max: 120** **Actual: 70**

### CODE REVIEW

File: 385/Assets/Scripts/RopeSystem.cs [hunk 1]
```diff
@@ -112,6 +112,13 @@ public class RopeSystem : MonoBehaviour
 
     void Start()
     {
+		// Set axis strings based on the user's controller selection from the menu screen
+		if (ChangeScene.SelectedControllerMode) 
+		{
+			FireAxis = "Fire_P2";
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/RopeSystem.cs]: We should have publicly accessible strings that can be changed in the unity editor at runtime, not hard coded strings.

File: 385/Assets/Scripts/PlayerController.cs [hunk 2]
```diff
@@ -118,6 +118,18 @@ public bool IsOnGround
     // Use this for initialization
     void Start ()
     {
+		// Set based on player's control selection on the menu screen
+		ControllerMode = ChangeScene.SelectedControllerMode;
+
+		// Set all the axis strings to use _P2 versions, enabling controller use
+		if (ControllerMode) 
+		{
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/PlayerController.cs]: See comment on RopeSystem.

File: 385/Assets/Scripts/ChangeScene.cs [hunk 3]
```diff
@@ -1,13 +1,20 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.UI;
 using UnityEngine.SceneManagement;
 
-public class ChangeScene : MonoBehaviour {
+public class ChangeScene : MonoBehaviour 
+{
+	public static bool SelectedControllerMode = false;
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/ChangeScene.cs]: We should not include this to ChangeScene. We should instead have a `GlobalSettings` gameobject that does not get unloaded on scene change, or something like it to handle settings instead.

Another group sent me this video.
https://youtu.be/J6FfcJpbPXE?t=4m7s

Controller mode is not relevant to change scene. Keep the separation of concerns in mind.

File: 385/Assets/Scripts/PlayerController.cs [hunk 4]
```diff
@@ -26,24 +26,40 @@ public class PlayerController : MonoBehaviour
 
     /// <summary>
     /// Which axis does the player use to strafe left and right on the ground, or adjust
-    /// their velocity when swinging?
+	/// their velocity when swinging? Set to either *_Mouse or *_Controller string in Start() based on
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/PlayerController.cs]: indenting?

File: 385/Assets/Scripts/PlayerController.cs [hunk 5]
```diff
@@ -26,24 +26,40 @@ public class PlayerController : MonoBehaviour
...
 
     /// <summary>
-    /// Which axis to check that indicates that the player wanted to jump
+	/// Which axis to check that indicates that the player wanted to jump. Set to either 
+	/// *_Mouse or *_Controller string in Start() based on user control preference.
     /// </summary>
-    public string JumpAxis = "Jump"; // use Jump_P2 for Player 2
+    private string JumpAxis; // use Jump_P2 for Player 2
+
+	public string JumpAxis_Mouse = "Jump";
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/PlayerController.cs]: This is good

**Comment by clayton-snyder** [comment]
All the requested changes should be addressed now. Let me know if you see anything that could be improved further.

**Pull request merged**